src/maker/rpc/server.rs

```diff
@@ -81,6 +81,9 @@ fn handle_request(maker: &Arc<Maker>, socket: &mut TcpStream) -> Result<(), Make
             let destination =
                 Destination::Address(Address::from_str(&address).unwrap().assume_checked());
 
+            // Sync the Wallet to ensure the UTXOs are updated
+            maker.get_wallet().write()?.sync_no_fail();
+
```

Comment:
I think its better to sync after the operation to get the updated utxo. As auto sync happens every 15 mins, we will have the utxo set updated at the start. Only after operation we should sync again to show the user updated balances.